cmd/nodecmd/sync.go

```diff
@@ -52,9 +59,80 @@ func syncSubnet(_ *cobra.Command, args []string) error {
 	if !isBootstrapped {
 		return errors.New("node is not bootstrapped yet, please try again later")
 	}
+	if err := checkAvalancheGoVersionCompatible(clusterName, subnetName); err != nil {
+		return err
+	}
 	return trackSubnet(clusterName, subnetName, models.Fuji)
 }
 
+func parseAvalancheGoOutput(fileName string) (string, error) {
+	jsonFile, err := os.Open(fileName)
+	if err != nil {
+		return "", err
+	}
+	defer jsonFile.Close()
+	byteValue, _ := io.ReadAll(jsonFile)
+
+	var result map[string]interface{}
+	if err = json.Unmarshal(byteValue, &result); err != nil {
+		return "", err
+	}
+	nodeIDInterface, ok := result["result"].(map[string]interface{})
+	if ok {
+		vmVersions, ok := nodeIDInterface["vmVersions"].(map[string]interface{})
+		if ok {
+			avalancheGoVersion, ok := vmVersions["platform"].(string)
```

Comment:
should we use result/version ?

Reply:
I mean, instead of using the version of vm platform in result/vmVersions/platform, to use the root version in results/version